### PR TITLE
Add provider zone option

### PR DESCRIPTION
**Pull request by dkorn**
It is possible to pass zone name for the added provider. The zone must already exist in manageiq.
If not passed, the 'default' zone will be chosen.

### CODE REVIEW

**Comment by dkorn** [comment]
@jaryn - please take a look at the changes I made to the tests.
@cben, @enoodle, @kwoodson, @simon3z please review

File: library/manageiq.py [hunk 1]
```diff
@@ -34,6 +34,11 @@
       - the openshift provider type
     required: true
     choices: ['openshift-origin', 'openshift-enterprise']
+  zone:
+    description:
+      - the provider zone in manageiq
```
Comment (cben) [inline library/manageiq.py]: zone name.

File: library/manageiq.py [hunk 2]
```diff
@@ -189,16 +205,17 @@ def add_or_update_provider(self, provider_name, provider_type, endpoints):
             and a short message describing the operation executed
         """
         message = ""
+        zone_id = self.find_zone_by_name(zone or 'default')
```
Comment (cben) [inline library/manageiq.py]: What if not found and we get zone_id `None`?
- Does update_provider deal with that?
- Does add_new_provider deal with that?

Or do all installations have a `default` zone?

**Review by cben — COMMENTED** [review]
Have you managed to test this on a real MIQ?

Perhaps add some test where zone is not the same?

File: library/manageiq.py (hunk 1 above)
Comment (dkorn) [reply library/manageiq.py]: right, I'll change this

File: library/manageiq.py (hunk 2 above)
Comment (dkorn) [reply library/manageiq.py]: IINM all installations have a default zone

**Comment by dkorn** [comment]
@enoodle, @kwoodson, @simon3z any comments?

**Comment by simon3z** [comment]
I gave just a quick look, LGTM

**Pull request merged**